x-pack/plugins/synthetics/server/synthetics_service/formatters/formatting_utils.ts

```diff
@@ -5,31 +5,101 @@
  * 2.0.
  */
 
+import { Logger } from '@kbn/logging';
+import { ParsedVars, replaceVarsWithParams } from './lightweight_param_formatter';
 import { ConfigKey, MonitorFields } from '../../../common/runtime_types';
+import variableParser from './variable_parser';
 
-type FormatterFn = (
-  fields: Partial<MonitorFields>,
-  key: ConfigKey
-) => string | null | Record<string, any> | string[];
+export type FormatterFn = (fields: Partial<MonitorFields>, key: ConfigKey) => string | null;
 
-export const arrayFormatter: FormatterFn = (fields, key) => {
+export const arrayToJsonFormatter: FormatterFn = (fields, key) => {
```

Comment:
Can we put these specific formatting utils in private formatters please? Most anything that is `toJSON` is a private formatter util.